### ISSUE TITLE
Magento collection condition SKU "is not one of" does not work

### ISSUE THREAD

**Issue by stasgrinML** [Smile-SA/elasticsuite#2521]
When you have a Magento widget that is using Widget Conditions fieldset and later on used on product collection by adding these SKU conditions with type "is not one of". However, Smile ES applies such an SKU filter under the "should" section which is completely opposite to what should happen, e.g. instead of filtering out these values, it forces values to be one of the SKUs listed.

![image](https://user-images.githubusercontent.com/73647419/162490989-4f2e8ef3-d586-4aca-9051-71fd94d1e751.png)

### Preconditions
Server: Nginx / PHP 7.4.27 / ES 7.16.3 / MySQL Percona 5.7.36
Magento Version: 2.4.3
ElasticSuite Version: 2.10.9.1
Environment: Developer mode for Magento
Third-party modules: this does not affect as the core Smile ES file is the issue

### Steps to reproduce
1. Create a new widget with conditions fieldset in it
```
    <parameter name="condition" xsi:type="conditions" visible="true" required="true" sort_order="10"
               class="Magento\CatalogWidget\Block\Product\Widget\Conditions">
        <label translate="true">Conditions</label>
    </parameter>
```
2. Create a Magento product collection:
```
$collection = $this->productCollectionFactory->create();
```
3. Load custom conditions:
 ```
// step 1: extract conditions
/**@var $conditonsHelper Magento\Widget\Helper\Conditions */
$conditions = $this->conditionsHelper->decode($conditions);
/**@var $rule Magento\CatalogWidget\Model\Rule */
$conditions = $this->rule->loadPost(['conditions' => $conditions])->getConditions();

// step 2: apply conditions to the collection
/**@var $sqlBuilder Magento\Rule\Model\Condition\Sql\Builder */
$this->sqlBuilder->attachConditionToCollection($collection, $conditions); // <-- THIS is where issue happens!
```

### Expected result
1. If the conditions set in a widget is "SKU is not one of ...." then the applied ES query filter will be under "must_not" section

### Actual result
1. ES applies "SKU is not one of ...." conditions with the "should" section, which messes up product collection

--

So what happens? I spent several hours debugging and found the issue. 
Basically, Smile ES has a special check for SKU conditions\filter being applied. So when you call `$this->sqlBuilder->attachConditionToCollection($collection, $conditions); ` Smile hooks in and rewrites default functionality.

File: `Smile\ElasticsuiteCatalogRule\Model\Rule\Condition\Product\QueryBuilder`
Line: 143
Code:

```
        if ($productCondition->getAttribute() === 'sku') {
            $queryParams = [];
            $fieldName   = $this->getSearchFieldName($productCondition);

            // SKU values can be an array of value, due to the picker.
            foreach (explode(',', $productCondition->getValue()) as $value) {
                // We add all of these values as a should match sub query.
                $queryParams['should'][] = $this->prepareQuery(    // <---------- THIS is the issue!
                    QueryInterface::TYPE_MATCH,
                    ['field' => $fieldName, 'queryText' => trim($value), 'minimumShouldMatch' => "100%"]
                );
            }

            // One clause must match between all.
            $queryParams['minimumShouldMatch'] = 1;
            $query = $this->prepareQuery(QueryInterface::TYPE_BOOL, $queryParams);
        }
```

As you can see, it checks if the filter attribute is SKU and if yes, simply dumps all SKUs under the "should" section with the "match 100%" option. However, when the filter is "is not one of", completely opposite should happen. The section must be "must_not"!

I have manually overridden this Smile ES class and added an if-clause to check actual operator, something like 
```
$productCondition->getOperator() == '!()' ? 'must_not' : 'should';
```
... and it worked totally fine. ES request now translates "SKU is not one of ..." into ES section "must_not" and product collection does match Magento collection with correct filters applied.

**Comment by romainruaud** [comment]
Hi @stasgrinML 

that's a good catch, I don't remember why we did consider that using SKU should alway be resolved as a "should" without taking the operator into account.

Can you propose a PR for this one ?

Regards

**Comment by stasgrinML** [comment]
@romainruaud unfortunately, I did a full class rewrite(di), not just a composer patch for it. I'll leave it with you from now on :)

**Comment by stasgrinML** [comment]
@romainruaud sorry for this offtopic for the original issue, but there is a mistype in code that is a much hassle to fix:
File: vendor/smile/elasticsuite/src/module-elasticsuite-core/Search/Request/SortOrder/SortOrderBuilder.php
Line: 82
Code:

`
    public function buildSordOrders(ContainerConfigurationInterface $containerConfig, array $orders)
`

Obviously, it should be sor*T* not sorD :)
But it has multiple mentions like this... so can't be fixed just in one file.

**Comment by romainruaud** [comment]
Unfortunately we'll not fix this one, because this is a public method that we know is often extended by merchants/integrators 

So we prefer to keep the mistype AND the backward-compatibility :)
